### PR TITLE
Feat/pupil 886/expandable quiz review item

### PR DESCRIPTION
**Pull request by benprotheroe**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

When I render a review item for quizzes,
I want to be able to nest extra information
So that I can render full quiz results.

## Link to the design doc
https://www.figma.com/design/G7z5ufo2T0CHk14CkadeuB/Pupil-Lesson?node-id=9597-174075&m=dev

## A link to the component in the deployment preview
https://deploy-preview-247--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-oaklessonreviewquiz--docs

## ACs

- [ ]  The item has a toggle button to expand
- [ ]  It expands with an area that can be populated by composition
- [ ]  mobile, tablet and desktop views
- [ ]  accessible

### CODE REVIEW

**Comment by messmike** [comment]
- For desktop, can we position the score to the right of the title, so that the button is the only item on the right-hand side: 
Screenshot 2024-07-29 at 9 34 47 am

- When expanded the space between the expanded white space and the items above is missing space (should be `space-between-m`): 
Screenshot 2024-07-29 at 9 40 09 am

- The chevron icon spins 360 when clicked/tapped. When the accordion is expanded the chevron icon should point up: 
Screenshot 2024-07-29 at 9 38 33 am

**Review by kimon-satan — CHANGES_REQUESTED** [review]
minor change to the header prop name

File: src/components/organisms/pupil/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 1]
```diff
@@ -60,21 +60,24 @@ export const OakLessonReviewItem = (props: OakLessonReviewItemProps) => {
...
-        $background={iconBackgroundColor}
-      />
-      <OakFlex $flexGrow={1} $flexShrink={1} $flexDirection={"column"}>
-        <OakBox $font={["heading-6", "heading-5"]} $color={"text-primary"}>
-          {pickLabelForSection(lessonSectionName)}
-        </OakBox>
-        <OakBox $font={["body-2", "body-1"]}>
-          {pickSummaryForProgress(props)}
-        </OakBox>
+      <OakFlex $gap="space-between-m" $alignItems="center" $width={"100%"}>
```
[the remark this answers is not in the document]
Comment (benprotheroe) [reply src/components/organisms/pupil/OakLessonReviewItem/OakPupilLessonReviewItem.tsx]: if you want to see how it works look at the code for the expandableReview component and the reviewItem components stories

**Review by messmike — APPROVED** [review]

**Review by kimon-satan — APPROVED** [review]

**Comment by RBrownsett** [comment]
@benprotheroe accessibility issue fixed - not sure if its a storybook quirk but when in small mobile view the quiz background does not span behind the text? 

image

**Comment by benprotheroe** [comment]
> @benprotheroe accessibility issue fixed - not sure if its a storybook quirk but when in small mobile view the quiz background does not span behind the text?
> 
> image

i think this is a storybook quirk

**Comment by RBrownsett** [comment]
@benprotheroe noticed that in tablet mode the score is aligned to the left - on the design its to the right

image

**Comment by benprotheroe** [comment]
> @benprotheroe noticed that in tablet mode the score is aligned to the left - on the design its to the right
> 
> image

yep, fixed

**Review by kimon-satan — APPROVED** [review]
nice one

**Pull request merged**